### ISSUE TITLE
how do you map the end?

### ISSUE THREAD

**Issue by MariusDarkblade** [overviewer/Minecraft-Overviewer#1917]
i suck at coding, all the docs are in python which i have 0 clue how to use. i use command prompt and use simple coding. how would i go about mapping out the end? do i have to use the DM1 folder when i select my source folder? i know DM1 is the end and DM-1 is the nether. i dont want to start typing in commands in command prompt without knowing fully what i am doing.

**Comment by MariusDarkblade** [comment]
![python excursion](https://user-images.githubusercontent.com/19295323/115808396-6a708100-a3b8-11eb-84bb-d636b2dbe93d.jpg)
so i tried out python thinking maybe i could figure out the docs and get something to work. i cant even get it to start. thought i got somewhere when i entered E:\ and i got ... instead of >>> but then nothing worked after that. i have no clue what i am doing.

**Comment by crlang44** [comment]
You will create a configuration files according to: http://docs.overviewer.org/en/latest/config/#config-file-specifications
In this configuration file you can specify a render which sets the "dimension" to "end"
Then you will call overviewer by providing this configuration file: `overviewer.py --config=path/to/my_configfile`

**Comment by drnixon** [comment]
Looks like you are trying to set the options manually with each render. That might work but it would be extra painful. The best method is to use a config file that tells Overviewer where your world lives, where you want the map saved, and what things you want it to render. You call this config file every time you run Overviewer.

Here's a very simple example config file, which renders overworld (day and night), nether, and end (end and nether using smooth lighting to make it easier to see the map). Copy the code and save it as "worldname.py" or whatever. Call it when running overviewer by running `/path/to/overviewer.py --config=/path/to/worldname.py`. (Note I'm on a Mac, adjust the path as needed for your platform.)

[example_config.txt](https://github.com/overviewer/Minecraft-Overviewer/files/6405670/example_config.txt)

There are lots of more advanced options you can use to make a config file like this more complicated to add more views, more options, map markers, player homes and locations, etc. You can find some of these options explained in the help documentation (config file explained here: [http://docs.overviewer.org/en/latest/config/]) and once you figure out some of the basics you might also take a look at what others have used and see if that inspires you to add tweaks to your own. I've had bases marked on my map for a long time, for example, but I recently saw a config file that did it more elegantly, so I edited my config to use that method.

**Comment by MariusDarkblade** [comment]
ok, i thank you guys for the help but at the same time....none of that made any sense to me. i have 0 ability to code, i dont know coding. the last bit of what you could probably call coding i did was back in high school 15 years ago when we built a webpage using html, and i barely remember how to do that now. i couldnt even get python to run overviewer, let alone try and get to any other point. the text document drnixon provided is great and all but what do i do with it? is it python or do i run it in command prompt?  i am going to assume "world name" should be replaced with the location of my world folder and "/path/to/world" should be the location i want the map to be but aside from that i have no clue what i am looking at. its great you guys are good at coding but any chance you can dumb it down a bit more, start at the beginning. the extent of my map rendering is opening up command prompt and following these code lines, which i copy paste into command prompt, 
E: 
cd E:\overviewer-0.16.12
overviewer.exe "C:\Users\(myname)\AppData\Roaming\.minecraft\saves\amped but not amped" "E:\MinecraftMaps\Amped but not amped"

thats all i know how to do. as shown in the picture above, i couldnt even get python to work so if you're going to tell me python script you gotta start at the beginning to get things running.

**Comment by drnixon** [comment]
So in your case "path to world" is `C:/Users(yourname)/AppData/Roaming/.minecraft/saves/amped but not amped` and "path to save" is `E:/MinecraftMaps/Amped but not amped`. I believe the path to the Minecraft jar file IS NOT needed on Windows, Overviewer should be able to find it so you ought to be able to delete that line from the config file entirely.

And yes, you must use FORWARD slashes (`/`) in the path names in the config file, not backslashes (`\`) as you normally would on Windows. (Python needs this, it's normal on UNIX/Linux/macOS, it's only weird for Windows users. Sorry.)

Take the config file, edit it to change the paths to the above. Change every instance of "world name" to "Amped but not amped" as that seems to be what you want the world to be called. Save the file, then rename it to (whatever you want).py - make sure it is a .py file and not a .txt file. If you don't see Windows asking you to confirm that you want to change the file type, you didn't successfully change it.

Your screenshot seems to suggest that you told your computer to open a Python prompt (via python.exe), first tried to run overviewer.exe, and then tried overviewer.py. You don't need to open Python first. Just run the overviewer.exe file directly, as a standalone file. (Your first attempt failed because Python can't run an .exe file, it can only run a .py file. Your second attempt failed because there IS no file in the Windows install download that is named "overviewer.py" - if you built from source you would see this, but the premade Windows package includes overviewer only as an .exe file.)

Go to a command prompt. CD into the folder where overviewer.exe lives. Copy your edited config script into this folder (for argument's sake let's assume you named the config "ampedbutnotamped.py"). Run the overviewer by entering:
`overviewer.exe --config=ampedbutnotamped.py`

If that doesn't work, plug in the FULL path names - e.g. `C:\folder_where_overviewer_lives\overviewer.exe --config=C:\folder_where_config_file_lives\ampedbutnotamped.py'

You might also look at the ["newbie" section](http://docs.overviewer.org/en/latest/win_tut/windowsguide/) in the documentation. You might find some tips there that could help.

**Comment by drnixon** [comment]
Also - don't be intimidated and don't be afraid to admit what you don't know. I'm not on Github because I'm a coder, I'm here because I use things that other people code. I know what I know because many other people were willing to patiently share their knowledge. I also know when I am in over my head - that's when I ask for help, just like you did here.

**Comment by MariusDarkblade** [comment]
so do i replace whats within the "_" or all of that? like where it says ["world name"] is the [" needed or do i get rid of the quotes to? also, i'm not intimidated or afraid to admit what i dont know, especially concerning this. i have 0 clue what i am doing and i have no problem saying that. my thing is this is basically all gibberish to me. so when someone says "yah you just do this" its like, ok....i do what now? calling me a clueless is actually an overstatement.

**Comment by MariusDarkblade** [comment]
![mapfail](https://user-images.githubusercontent.com/19295323/116728321-c12b1b80-a9b3-11eb-9c5c-72f6c606f3df.png)
ok i think i did something wrong

**Comment by MariusDarkblade** [comment]
i just tried the full path as well and it didnt work. i thought maybe the spaces in the file name caused the problem so i got rid of them but now instead of saying its not a recognized command its saying invalid syntax.

**Comment by MariusDarkblade** [comment]
![map succeed](https://user-images.githubusercontent.com/19295323/116730781-e53c2c00-a9b6-11eb-90ff-725dcba7ef5a.jpg)
i think i got it. tweaked it a bit after redoing the py file. hopefully this works.

**Comment by MariusDarkblade** [comment]
yah, it generated everything. took up a lot more space on my hard drive than i expected but i have plenty to work with. so, python, as far as working through windows is concerned, works better when you create a py file using notepad and then running it through command prompt? my original thought process when i tried creating the map using python itself was i would have to type each line of code one by one. i thought it would just read the files.

**Comment by drnixon** [comment]
You could type it all in using a single command, but the overviewer script looks for extra options ("arguments") when you run it. One of these arguments tells it "Hey, I have a file right here with a list of things I'd like you to do when running this script" - it saves you a whole lot of time not having to enter that every time you want to render your world. When you run overviewer.exe and include the folder where your world is and the folder where you want the map, you are also passing arguments (world location and map save location). 

The Windows package includes Python as a program file (python.dll), plus the overviewer script as an executable file (overviewer.exe). As far as I can tell you don't even need to have Python installed to use this, the pre-packaged python.dll has everything needed to make it work. (Which is much, MUCH easier than having to build it from source as you do on macOS.)

If you are running any python scripts directly, saving them in notepad as "script_name.py" means you can run them by typing in "script_name.py" in a command line (faster than typing python.exe and then entering the info one line at a time at the >>> prompt!). If python is installed, Windows knows that .py files should be opened with python. When you ran overviewer.exe, you were using the pre-packaged python.dll in the overviewer folder, not the python program that you installed on your machine.

**Comment by MariusDarkblade** [comment]
ah ok. still way above me but i think i'm starting to get some of it.
